test/test.test.ts

```diff
@@ -0,0 +1,316 @@
+    })
+
+    it('should pass if filter is passed and no options is passed', () => {
+      expect(() => { validateParameters('some test', emptyOptions) }).not.to.throw()
+    })
+
+    it('should pass if options is passed and no filter is passed', () => {
+      expect(() => { validateParameters(undefined, {
+        ...emptyOptions,
+        test: 'some test',
+      }) }).not.to.throw()
+    })
+
+    it('should fail if filter and options are passed', () => {
+      expect(() => { validateParameters('some describe', {
+        ...emptyOptions,
+        test: 'some test',
+      }) }).to.throw(/You should either use filter by full name or file/)
+    })
+
+  })
+
+  describe('tabulations for node', () => {
+
+    it('should work for root package', () => {
+      expect(tabulationForNode({ fullyQualifiedName: 'root' })).to.equal('')
+    })
+
+    it('should work for fqn level', () => {
+      expect(tabulationForNode({ fullyQualifiedName: 'aves.otros.pepita' })).to.equal('    ')
+    })
+  })
+
+  describe('smoke test for test default function', () => {
+
+    let loggerInfoSpy: sinon.SinonStub
+    let processExitSpy: sinon.SinonStub
+
+    const projectPath = join('examples', 'test-examples', 'normal-case')
+
+    const emptyOptions = {
+      project: projectPath,
+      skipValidations: false,
+      file: undefined,
+      describe: undefined,
+      test: undefined,
+    }
+
+    beforeEach(() => {
+      loggerInfoSpy = sinon.stub(logger, 'info')
+      processExitSpy = sinon.stub(process, 'exit')
+    })
+
+    afterEach(() => {
+      sinon.restore()
+    })
+
+    it('passes all the tests successfully and exits normally', async () => {
+      await test(undefined, {
+        ...emptyOptions,
+        file: 'test-one',
+      })
+
+      expect(processExitSpy.callCount).to.equal(0)
+      expect(spyCalledWithSubstring(loggerInfoSpy, 'Running 2 tests')).to.be.true
+      expect(spyCalledWithSubstring(loggerInfoSpy, '2 passing')).to.be.true
+      expect(spyCalledWithSubstring(loggerInfoSpy, '0 failing')).to.be.false // old version
+    })
+
+    it('returns exit code 2 if one or more tests fail', async () => {
+      await test(undefined, emptyOptions)
+
+      expect(processExitSpy.calledWith(2)).to.be.true
+      expect(spyCalledWithSubstring(loggerInfoSpy, 'Running 5 tests')).to.be.true
+      expect(spyCalledWithSubstring(loggerInfoSpy, '4 passing')).to.be.true
+      expect(spyCalledWithSubstring(loggerInfoSpy, '1 failing')).to.be.true
+    })
+
+    it('returns exit code 2 if one or more tests fail', async () => {
+      await test(undefined, emptyOptions)
+
+      expect(processExitSpy.calledWith(2)).to.be.true
+      expect(spyCalledWithSubstring(loggerInfoSpy, 'Running 5 tests')).to.be.true
+      expect(spyCalledWithSubstring(loggerInfoSpy, '4 passing')).to.be.true
+      expect(spyCalledWithSubstring(loggerInfoSpy, '1 failing')).to.be.true
+    })
```

Comment:
```suggestion
    it('returns exit code 2 if one or more tests fail', async () => {
      await test(undefined, emptyOptions)

      expect(processExitSpy.calledWith(2)).to.be.true
      expect(spyCalledWithSubstring(loggerInfoSpy, 'Running 5 tests')).to.be.true
      expect(spyCalledWithSubstring(loggerInfoSpy, '4 passing')).to.be.true
      expect(spyCalledWithSubstring(loggerInfoSpy, '1 failing')).to.be.true
    })
```

Creo que te quedó duplicado

Reply:
uy sí! tenés razón!
uy sí! tenés razón!
😄